yaml/converter/legacy/internal/config.go

```diff
@@ -65,12 +69,41 @@ func Convert(d []byte) ([]byte, error) {
 			toContainer(container),
 		)
 	}
+
 	pipeline.Volumes = toVolumes(from)
 	pipeline.Trigger.Branch.Include = from.Branches.Include
 	pipeline.Trigger.Branch.Exclude = from.Branches.Exclude
 
-	manifest := &droneyaml.Manifest{}
-	manifest.Resources = append(manifest.Resources, pipeline)
+	if len(from.Matrix.Include) > 0 {
+		for _, environ := range from.Matrix.Include {
+			current := pipeline
+			current.Name = fmt.Sprintf("default-%x", structhash.Sha1(environ, 1))
```

Comment:
I think the length of the generated pipeline names is going to be an issue. What about instead using a counter, like `name: matrix1`, `name: matrix2`, etc?

Reply:
Haven't thought about something simple like that... Of course, that should work.

Reply:
or maybe something like this:
```
		for i, environ := range from.Matrix.Include {
			current := pipeline
			current.Name = fmt.Sprintf("%s-i", environ)
```

that way you end up with something like `golang-1`, `golang-2`, which might be more descriptive than `default`.

Reply:
Now it's simply a naming like `matrix-1`, `matrix-2` and so on :)

Reply:
`environ` is a map of env variables, maps are randomly ordered which makes it hard to test since the ordering can always change :)

Reply:
instead of iterating the map, maybe add the map keys to a []string and sort it, and then iterate on the sorted list?

Reply:
Now I'm just iterating through environ to replace the variables